framework/filters/AccessRule.php

```diff
@@ -161,8 +191,13 @@ protected function matchRole($user)
                 if (!$user->getIsGuest()) {
                     return true;
                 }
-            } elseif ($user->can($role)) {
-                return true;
+            } else {
+                if (!isset($roleParams)) {
```

Comment:
lets say i have `images` which any auth user ('@') can upload but only the owner can delete them.

or anyone can upload images anonymously ('?') but only the creator ip can delete them?